keycloak-service-module/src/main/java/org/aerogear/mobile/keycloak_service_module/KeyCloakService.java

```diff
@@ -28,30 +25,34 @@
     private String subjectTokenType;
     private String requestedTokenType;
     private String realm;
-    private MobileCore core;
     private String resource;
     private HttpServiceModule httpModule;
     private String accessToken;
 
-
-    public KeyCloakService() {
+    private KeyCloakService(KeyCloakServiceConfiguration keyCloakServiceConfiguration) {
+        this.serverUrl = keyCloakServiceConfiguration.serverUrl;
+        this.clientId = keyCloakServiceConfiguration.clientId;
+        this.audience = keyCloakServiceConfiguration.audience;
+        this.grantType = keyCloakServiceConfiguration.grantType;
+        this.subjectTokenType = keyCloakServiceConfiguration.subjectTokenType;
+        this.requestedTokenType = keyCloakServiceConfiguration.requestedTokenType;
+        this.realm = keyCloakServiceConfiguration.realm;
+        this.resource = keyCloakServiceConfiguration.resource;
+        this.httpModule = keyCloakServiceConfiguration.httpModule;
     }
 
-
-    @Override
-    public void bootstrap(MobileCore core, ServiceConfiguration config) {
-        this.serverUrl = config.getProperty("auth-server-url");
-        this.clientId = config.getProperty("clientId");
-        this.audience = config.getProperty("audience");
-        this.grantType = config.getProperty("grant_type");
-        this.subjectTokenType = config.getProperty("subject_token_type");
-        this.requestedTokenType = config.getProperty("requested_token_type");
-        this.resource = config.getProperty("resource");
-        this.realm = config.getProperty("realm");
-        this.core = core;
-        this.httpModule = (HttpServiceModule) core.getService("http");
+    public static KeyCloakService getInstance() {
+        ServiceConfiguration config = MobileCore.getServiceConfiguration("keycloak");
```

Comment:
So one of the things we have been talking about is having multiple instance of the same service type, so I'm a bit iffy on this section right here.  If we have multiple "keycloak typed" services this won't resolve and there isn't a fallback way to instanciate them by name.